### DISCUSSION TITLE
Hide somethings & Gallery short code

### DISCUSSION THREAD

**Discussion by nevikw39** [razonyang/hugo-theme-bootstrap#582] [General]
您好，

我已經把 `params.toml` 中的 `palettes` 設成 `[]`, 但是右上方還是會顯示空的選單。

另外，是否可以移除切換 light, dark 的按鈕呢？？正文跟 sidebar 中間的 post-panel 我也想要隱藏起來。

使用 `{{< gallery images="..." >}}` 時圖片的比例被壓得很奇怪，請問要怎麼處理？？

感謝！！

**Comment by razonyang** [comment]
1. `palettes` 的問題已經修復了，為空時不顯示相關按鈕。
2. 可以通過設置 `post.panel` 為 `false` 以隱藏 post panel（最新的commit）。
3. 至於深色和亮色的切換按鈕 ，你是只保留 auto 選項嗎？還是說去掉整個切換按鈕？
4. 而 Gallery 的話，缺少例子，對這個問題缺少直觀的感受，不好確定。

**Comment by nevikw39** [comment]
前兩個問題已經解決了，感謝

第三個的話就是希望整個都移除，因為我的網站搜尋欄右邊沒有切換語言或顏色的按鈕，整個移除掉比較不突兀

最後 gallery 的話我這邊 Safari, Chrome 的狀況都類似：
![image](https://user-images.githubusercontent.com/20489759/176745377-bcb67405-b6ba-4cf1-8d9f-d27d105d9449.png)

**Comment by razonyang** [comment]
1. 現在可以通過設置 `topAppBar.colorToggle` 為 `false` 禁用該顏色開關
2. 至於 Gallery 的話，沒有復現出來，我試了多個不同比例的圖片，效果如下：
![image](https://user-images.githubusercontent.com/17720932/176824961-cea158c6-fc98-47c5-bd24-bb7853bcd7eb.png)
> 不過我做了一些小修復，對於 Gallery 的圖片不進行 lazy loading，否則會出現圖片重疊的問題。

如果更新後，還是出現類似問題，可以將你的圖片發到我的郵箱或者這裡，以便復現和解決。

**Reply by nevikw39** [reply]
`colorToggle` 的部分沒有問題，gallery short code 我是 `params.toml` 中全域啟用 masonry, 語法是` {{< gallery images="images/abpc.jpeg,images/hpcw.jpeg" >}}`

兩張相片在此，一張是垂直的一張是水平的，`<img>` 的 `width`, `height` 屬性分別被顯式得設為 960, 1280 及 1280, 960:
![abpc](https://user-images.githubusercontent.com/20489759/176842339-f2f5c162-cd92-4cd4-8f4e-cce322225dbc.jpeg)
![hpcw](https://user-images.githubusercontent.com/20489759/176842262-d77da8a3-ddb5-4124-bae8-348ad087875e.jpeg)

**Reply by razonyang** [reply]
通過 CSS 解決了，更新到最新 commit 或者 master 分支即可，後續我再看看如何優化。

**Comment by nevikw39** [comment]
Gallery 的部分現在也看起來好極了！！

另外想請問一下，我這邊 theme 的上一個版本是 2/5 的 `395f54a`, 升到最近的版本一開始 hugo 發生一些錯誤，先把 hugo 從 0.91.2 upgrade to 0.101.0, 接著又執行 `hugo mod npm pack`, `npm install` 等等，`config.toml` 必須加入 `[build] writeStats = true`, 目前是可以在 local 執行 serve, 但我有點好奇這部分的差異。

我是用 GitHub Actions 來 build 的，所以 `hugo mod npm pack` 也必須加到 workflows 中嗎？？那麼 `npm install` 是必須的嗎？？

現在 repo 目錄下多了 `.hugo_build.lock`, `assets/jsconfig.json`, `hugo_stats.json` 等檔案，不知道是否需要 commit?? `resources/_gen` 目錄底下也出現一些檔案。

另外比較大的問題是過往我有 `assets/css/custom.css`, `assets/js/custom.js`, 那時不必在設定 `params.toml` 也有效，但現在都沒有作用。

問題有點零散，先感謝您！！

**Reply by razonyang** [reply]
1. `hugo mod npm pack` 不是必須的，其作用是拉取和合併主題的依賴到你的項目中去，每次升級後都需要進行這個操作，並提交相應的 package.json 和 package-lock.json
2. `npm install` 是必須的，因為構建站點需要的 `node_modules` 一般不會提交到項目倉庫。
3. 而其他 .hugo_build.lock, assets/jsconfig.json, hugo_stats.json, resources 等都是站點構建時產生的文件，也不需要提交。
4. 至於 assets/css/custom.css, assets/js/custom.js，是因為很久之前（`v0.68.0`）做了大改動，你只需要分別重命名為 assets/main/scss/_custom.scss 和 assets/main/js/custom.ts 即可。具體的影響和改動可以參閱 https://github.com/razonyang/hugo-theme-bootstrap/blob/master/UPGRADE.md

**Reply by razonyang** [reply]
你也可以參考我使用 Docker 構建的命令：https://github.com/razonyang/blog/blob/main/Dockerfile

1. npm install
2. npm install -g @fullhuman/postcss-purgecss rtlcss
3. hugo...

**Reply by nevikw39** [reply]
了解了，印象中我之前都沒有 `npm install` 也能建置成功

**Reply by nevikw39** [reply]
那麼 `package-lock.json`, `package.hugo.json`, `package.json` 是不是也可以 ignore??

因為他們應該是在 `hugo mod npm pack`, `npm i` 時產生

**Reply by razonyang** [reply]
這些是必須的，npm install 會用到。

最初的設計是我本地構建好 JS 和 CSS 上傳的，所以當時是不需要自行構建，不過沒法移除用戶不需要的 CSS。

**Reply by nevikw39** [reply]
OK~~
